independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -568,35 +565,21 @@ boolean isNormalScope(Class<? extends Annotation> annotationType) {
         // Try to resolve the ambiguity
         List<InjectableBean<?>> resolved = new ArrayList<>(matching);
 
-        for (Iterator<InjectableBean<?>> iterator = resolved.iterator(); iterator.hasNext();) {
-            InjectableBean<?> beanInfo = iterator.next();
-            if (beanInfo.isDefaultBean()) {
-                iterator.remove();
-            }
-        }
+        resolved.removeIf(InjectableBean::isDefaultBean);
         if (resolved.size() == 1) {
             return Collections.singleton(resolved.get(0));
         }
 
-        for (Iterator<InjectableBean<?>> iterator = resolved.iterator(); iterator.hasNext();) {
-            InjectableBean<?> bean = iterator.next();
-            if (bean.getAlternativePriority() == null
-                    && (bean.getDeclaringBean() == null || bean.getDeclaringBean().getAlternativePriority() == null)) {
-                // Remove non-alternatives
-                iterator.remove();
-            }
-        }
+        // Remove non-alternatives
+        resolved.removeIf(bean -> bean.getAlternativePriority() == null
```

Comment:
So in general we try to keep the number of lambdas in runtime modules minimal. But in this particular case, it might make sense to add a `private static boolean isAlternativeOrDeclaredOnAlternative(InjectableBean bean)` method to `ArcContainerImpl` and use a method reference to make it more readable. WDYT?

Reply:
I agree. I'll create the method.